File: app/taskMgr/src/main/java/org/exthmui/share/taskMgr/entities/GroupEntity.java
package org.exthmui.share.taskMgr.entities;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.exthmui.share.taskMgr.Group;

import java.util.ArrayList;
import java.util.List;

@Entity(tableName = "group")
public class GroupEntity {
    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "group_id")
    public final String groupId;
    @ColumnInfo(name = "max_concurrent_tasks")
    public int maxConcurrentTasks;
    @ColumnInfo(name = "task_ids")
    public final List<String> taskIds = new ArrayList<>();

    public GroupEntity(@NonNull String groupId, int maxConcurrentTasks, List<String> taskIds) {
        this.groupId = groupId;
        this.maxConcurrentTasks = maxConcurrentTasks;
        this.taskIds.addAll(taskIds);
    }

    public GroupEntity(@NonNull Group group) {
        this.groupId = group.getGroupId();
        updateGroup(group);
    }

    public void updateGroup(Group group) {
        maxConcurrentTasks = group.getMaxConcurrentTasks();
        taskIds.clear();
        taskIds.addAll(group.getTaskIds());
    }

}

File: app/taskMgr/src/main/java/org/exthmui/share/taskMgr/entities/GroupEntity.kt
package org.exthmui.share.taskMgr.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import org.exthmui.share.taskMgr.Group
import org.exthmui.share.taskMgr.converters.StringListConvertor

@Entity(tableName = "group")
class GroupEntity {
    @PrimaryKey
    @ColumnInfo(name = "group_id")
    val groupId: String

    @ColumnInfo(name = "max_concurrent_tasks")
    var maxConcurrentTasks = 0

    @ColumnInfo(name = "task_ids")
    @TypeConverters(StringListConvertor::class)
    val taskIds: MutableList<String> = ArrayList()

    constructor(groupId: String, maxConcurrentTasks: Int, taskIds: MutableList<String>) {
        this.groupId = groupId
        this.maxConcurrentTasks = maxConcurrentTasks
        this.taskIds.addAll(taskIds)
    }

    constructor(group: Group) {
        groupId = group.groupId
        updateGroup(group)
    }

    fun updateGroup(group: Group) {
        maxConcurrentTasks = group.maxConcurrentTasks
        taskIds.clear()
        taskIds.addAll(group.getTaskIds())
    }
}